independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/UnwrapException.java

```diff
@@ -0,0 +1,23 @@
+package org.jboss.resteasy.reactive.server;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Used to configure that an exception (or exceptions) should be unwrapped during exception handling.
+ * <p>
+ * Unwrapping means that when an {@link Exception} of the configured type is thrown and no
+ * {@code jakarta.ws.rs.ext.ExceptionMapper} exists,
```

Comment:
This is a bit subtle. This unwrapping only happens when there's no registered mapper for the direct exception type?

Reply:
Correct

Reply:
Do you think this should be mentioned in the docs? (The guide, not just the javadoc)

Reply:
It is implied, no?

Reply:
I did not find it implied, no. How I read it was that the minute we declare that an exception should be unwrapped, it will never match an exceptoin mapper.

Reply:
I'm not sure how to phrase this in the docs, especially given that it's already a TIP followed by a NOTE, I feel like this succession of tips/notes should be turned into sub-sections :-/

Also, I'm now realising that it should probably be a warning (or error?) if we have both an exception mapper for `T` and an unwrap exception for `T`, no? In this case, the exception will never be unwrapped.

Reply:
> Also, I'm now realising that it should probably be a warning (or error?) if we have both an exception mapper for T and an unwrap exception for T, no? In this case, the exception will never be unwrapped.

Yeah, that can be a good follow-up